src/capital-project/capital-project.repository.ts

```diff
@@ -0,0 +1,62 @@
+import { Inject } from "@nestjs/common";
+import { isNotNull, sql } from "drizzle-orm";
+import { DataRetrievalException } from "src/exception";
+import { FindCapitalProjectTilesPathParams } from "src/gen";
+import { DB, DbType } from "src/global/providers/db.provider";
+import { capitalProject } from "src/schema";
+import { FindTilesRepo } from "./capital-project.repository.schema";
+
+export class CapitalProjectRepository {
+  constructor(
+    @Inject(DB)
+    private readonly db: DbType,
+  ) {}
+
+  async findTiles(
+    params: FindCapitalProjectTilesPathParams,
+  ): Promise<FindTilesRepo> {
+    const { z, x, y } = params;
+    try {
+      const tile = this.db
+        .select({
+          managingCodeCapitalProjectId:
+            sql<string>`${capitalProject.managingCode} || ${capitalProject.id}`.as(
+              `managingCodeCapitalProjectId`,
+            ),
+          managingAgency: sql`${capitalProject.managingAgency}`.as(
+            `managingAgency`,
+          ),
+          geom: sql<string>`
+            CASE 
+              WHEN ${capitalProject.mercatorFillMPoly} && ST_TileEnvelope(${z},${x},${y})
```

Comment:
Maybe a nit pick but could `WHEN ${capitalProject.mercatorFillMPoly}` be more specific? If I'm understanding the SQL, it's checking for that column to be _falsey_ as opposed to specifically `null`.

Reply:
The `&&` is a PostGIS [geometry overlap](https://postgis.net/docs/geometry_overlaps.html) check rather than a JS [logical AND](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND) check.

It's checking that the capital project geometry is within the specified envelope